onebusaway-android/src/main/java/org/onebusaway/android/util/UIUtils.java

```diff
@@ -1447,13 +1448,24 @@ public static void closeKeyboard(Context context, View v) {
             allIds.add(s.getId());
         }
 
+        // Do the same for filtered routes
+        HashSet<String> filterIds = new HashSet<>();
+        if (filter != null) {
+            for (String routeId : filter) {
+                filterIds.add(routeId);
+            }
+        }
+
         // Scan through the routes, and if a route-specific situation hasn't been added yet, add it
+        // If a filter list exists and a route isn't on it, skip it's situations
```

Comment:
Change to `If a filter list exists and a route_id is not included in the filter list, don't include it's situations in the returned list.`